sdk/dotnet/Pulumi/Core/Input.cs

```diff
@@ -75,5 +75,25 @@ public static class InputExtensions
 
         public static Output<T> ToOutput<T>(this Input<T>? input)
             => input ?? Output.Create(default(T)!);
+        
+        public static void Add<T, U>(this InputMap<Union<T, U>> map, string key, Input<T> value)
+        {
+            map.Add(key, value.ToOutput().Apply(v => (Union<T, U>)v));
+        }
+        
+        public static void Add<T, U>(this InputMap<Union<T, U>> map, string key, Input<U> value)
+        {
+            map.Add(key, value.ToOutput().Apply(v => (Union<T, U>)v));
+        }
+        
+        public static void Add<T, U>(this InputList<Union<T, U>> map, Input<T> value)
+        {
+            map.Add(value.ToOutput().Apply(v => (Union<T, U>)v));
+        }
+        
+        public static void Add<T, U>(this InputList<Union<T, U>> map, Input<U> value)
+        {
+            map.Add(value.ToOutput().Apply(v => (Union<T, U>)v));
+        }
```

Comment:
I could also imagine creating separate `InputMapExtensions` and `InputListExtensions` classes, but don't see a ton of value/need to do that, so I'm fine with them being on `InputExtensions`.

Reply:
Yeah, I was on a fence there. Do you know of any best practices/conventions around that?

Reply:
After reading some more, I went ahead and split it.